### PR TITLE
update make files with dependecies for parrallel building

### PR DESCRIPTION
**Pull request by jinghe-INTC**
To fix the problem found in https://github.com/intel/intel-sgx-ssl/issues/65

### CODE REVIEW

File: Linux/sgx/test_app/sgx_t.mk [hunk 1]
```diff
@@ -152,15 +151,15 @@ $(ENCLAVE_DIR)/TestEnclave_t.o: $(ENCLAVE_DIR)/TestEnclave_t.c
 	$(VCC) $(TestEnclave_C_Flags) -c $< -o $@
 	@echo "CC   <=  $<"
 
-$(ENCLAVE_DIR)/%.o: $(ENCLAVE_DIR)/%.cpp
+$(ENCLAVE_DIR)/%.o: $(ENCLAVE_DIR)/%.cpp $(ENCLAVE_DIR)/TestEnclave_t.o
```
Comment (lzha101) [inline Linux/sgx/test_app/sgx_t.mk:154]: Why use _t.o?  Suppose the dependency is the generated header file. So using _t.c is better.

File: Linux/sgx/test_app/sgx_u.mk [hunk 2]
```diff
@@ -118,7 +118,7 @@ $(UNTRUSTED_DIR)/TestEnclave_u.o: $(UNTRUSTED_DIR)/TestEnclave_u.c
 	$(VCC) $(App_C_Flags) -c $< -o $@
 	@echo "CC   <=  $<"
 
-$(UNTRUSTED_DIR)/%.o: $(UNTRUSTED_DIR)/%.cpp
+$(UNTRUSTED_DIR)/%.o: $(UNTRUSTED_DIR)/%.cpp $(UNTRUSTED_DIR)/TestEnclave_u.o
```
Comment (lzha101) [inline Linux/sgx/test_app/sgx_u.mk:121]: Similar comment. Using _u.c should be bettter.

File: Linux/sgx/test_app/sgx_t.mk (hunk 1 above)
Comment (jinghe-INTC) [reply Linux/sgx/test_app/sgx_t.mk]: I agree with you. Updated.

File: Linux/sgx/test_app/sgx_u.mk (hunk 2 above)
Comment (jinghe-INTC) [reply Linux/sgx/test_app/sgx_u.mk]: I agree with you. Updated.

**Pull request merged**